batch/src/main/java/org/openmrs/analytics/FhirSearchUtil.java

```diff
@@ -0,0 +1,113 @@
+package org.openmrs.analytics;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.parser.IParser;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.client.utils.URLEncodedUtils;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Bundle.BundleEntryComponent;
+import org.hl7.fhir.r4.model.Bundle.BundleLinkComponent;
+import org.hl7.fhir.r4.model.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FhirSearchUtil {
+  private static final Logger log = LoggerFactory.getLogger(FhirSearchUtil.class);
+
+  private FhirContext fhirContext;
+  private FhirStoreUtil fhirStoreUtil;
+
+  FhirSearchUtil(FhirStoreUtil fhirStoreUtil, FhirContext fhirContext) {
+    this.fhirStoreUtil = fhirStoreUtil;
+    this.fhirContext = fhirContext;
+  }
+
+  Bundle searchForResource(String baseSearchUrl, int count, int first,
+      String jsessionId, boolean parse, FhirContext fhirContext) {
+    try {
+      // TODO switch to HAPI FHIR client if all required features are supported.
+      URIBuilder uriBuilder = new URIBuilder(baseSearchUrl);
+      HttpUriRequest request = RequestBuilder
+          .get()
+          .setUri(uriBuilder.build())
+          .addHeader("Content-Type", "application/fhir+json")
+          .addHeader("Accept-Charset", "utf-8")
+          .addHeader("Accept", "application/fhir+json")
+          .addHeader("Cookie", "JSESSIONID=" + jsessionId)
+          .addParameter("_count", String.valueOf(count))
```

Comment:
are the `_count`, `_getpagesoffset`, and `_summary` params currently supported by the FHIR2 module?

Reply:
Yes and it made this approach much simpler to implement.